daemon/vpn/nordlynx/libtelio/libtelio.go

```diff
@@ -103,160 +125,89 @@ type Libtelio struct {
-	BoringtunResetConnections       bool                  `json:"boringtun_reset_connections,omitempty"`
-	FlushEventsOnStopTimeoutSeconds int                   `json:"flush_events_on_stop_timeout_seconds,omitempty"`
-	PostQuantumVPN                  *postQuantumVPNConfig `json:"post_quantum_vpn,omitempty"`
-	LinkDetection                   *linkDetectionConfig  `json:"link_detection,omitempty"`
-}
-
-type lanaConfig struct {
-	EventPath string `json:"event_path"`
-	Prod      bool   `json:"prod"`
-}
-
-type directConfig struct {
-	EndpointIntervalSecs  int                          `json:"endpoint_interval_secs,omitempty"`
-	SkipUnresponsivePeers *skipUnresponsivePeersConfig `json:"skip_unresponsive_peers,omitempty"`
-	Providers             []string                     `json:"providers,omitempty"`
-}
-
-type skipUnresponsivePeersConfig struct {
-	NoRxThresholdSecs int `json:"no_rx_threshold_secs,omitempty"`
-}
-
-type nurseConfig struct {
-	Fingerprint                string     `json:"fingerprint"`
-	HeartbeatInterval          int        `json:"heartbeat_interval,omitempty"`
-	InitialHeartbeatInterval   int        `json:"initial_heartbeat_interval,omitempty"`
-	EnableNatTypeCollection    bool       `json:"enable_nat_type_collection,omitempty"`
-	EnableRelayConnData        bool       `json:"enable_relay_conn_data,omitempty"`
-	EnableNatTraversalConnData bool       `json:"enable_nat_traversal_conn_data,omitempty"`
-	Qos                        *qosConfig `json:"qos,omitempty"`
-}
-
-type qosConfig struct {
-	RttInterval int      `json:"rtt_interval,omitempty"`
-	RttTries    int      `json:"rtt_tries,omitempty"`
-	RttTypes    []string `json:"rtt_types,omitempty"`
-	Buckets     int      `json:"buckets,omitempty"`
-}
-
-type derpConfig struct {
-	TcpKeepalive               int  `json:"tcp_keepalive,omitempty"`
-	DerpKeepalive              int  `json:"derp_keepalive,omitempty"`
-	EnablePolling              bool `json:"enable_polling,omitempty"`
-	UseBuiltInRootCertificates bool `json:"use_built_in_root_certificates,omitempty"`
-}
-
-type wireguardConfig struct {
-	PersistentKeepAlive *persistentKeepAliveConfig `json:"persistent_keepalive,omitempty"`
-}
-
-type persistentKeepAliveConfig struct {
-	Proxying int `json:"proxying,omitempty"`
-	Direct   int `json:"direct,omitempty"`
-	Vpn      int `json:"vpn,omitempty"`
-	Stun     int `json:"stun,omitempty"`
-}
-
-type postQuantumVPNConfig struct {
-	HandshakeRetryIntervalS int `json:"handshake_retry_interval_s,omitempty"`
-	RekeyIntervalS          int `json:"rekey_interval_s,omitempty"`
-}
-
-type linkDetectionConfig struct {
-	RttSeconds int `json:"rtt_seconds,omitempty"`
-}
-
-func handleTelioConfig(eventPath, deviceID, version string, prod bool, vpnLibCfg vpn.LibConfigGetter) ([]byte, error) {
-	telioConfig := &telioFeatures{}
+func handleTelioConfig(eventPath, version string, prod bool, vpnLibCfg vpn.LibConfigGetter) (*teliogo.Features, error) {
```

Comment:
`deviceID` is no longer needed in libtelio `v5.0.0-rc3` (it's taken by libtelio from the context)

Reply:
Which code/logic part does put `deviceID` into context?

Reply:
Sorry for the late response. The `deviceID` is set on moose initialization [here](https://github.com/NordSecurity/nordvpn-linux/blob/LVPN-3365-dynamic-linking/events/moose/moose.go#L168). I also confirmed it in `moose.db` and it is being set